include/bluetooth/bluetooth.h

```diff
@@ -404,6 +537,141 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
  */
 int bt_le_adv_stop(void);
 
+/** @brief Create advertising set.
+ *
+ *  Create a new advertising set and set advertising parameters.
+ *
+ *  @param[in] param Advertising parameters.
+ *  @param[in] cb    Callback struct to notify about advertiser activity. Can be
+ *                   NULL. Must point to valid memory during the lifetime of the
+ *                   advertising set.
+ *  @param[out] adv  Valid advertising set object on success or NULL otherwise.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_ext_adv_create(const struct bt_le_adv_param *param,
+			 const struct bt_le_ext_adv_cb *cb,
+			 struct bt_le_ext_adv **adv);
+
+struct bt_le_ext_adv_start_param {
+	/** Advertiser timeout (N * 10 ms), Application will be notified by the
+	 *  advertiser sent callback. Set to zero for no timeout.
+	 *
+	 *  If privacy :option:`CONFIG_BT_PRIVACY` is enabled then the timeout
+	 *  must be less than :option:`CONFIG_BT_RPA_TIMEOUT`.
+	 */
+	u16_t timeout;
+	/** Number of advertising events. Application will be notified by the
+	 *  advertiser sent callback. If used together with timeout then
+	 *  the condition that happens first will stop the advertiser.
+	 *  Set to zero for no limit.
+	 */
+	u8_t  num_events;
+};
+
+/** Start advertising with the given advertising set
+ *
+ *  Start advertising with a specific advertising set.
+ *  If both timeout and advertising event limit is given then the advertising
```

Comment:
Is it intentional that the application must set a duration both in terms of time and number of events? The HCI specification allows the application to only set one of these

Reply:
There is no must here, the application can set it to zero.

Reply:
I guess this was me interpreting this sentence a bit different. I would rather use the word "either" instead of "both" as it refers to giving a limit or not.

Reply:
Ok. Rephrased this. Hopefully it is clearer now. I wanted to explain that it is the first limit that stops the advertise (not that both must be reached).